src/jit/compiler.cpp

```diff
@@ -4272,10 +4293,14 @@ void Compiler::compFunctionTraceEnd(void* methodCodePtr, ULONG methodCodeSize, b
         {
             printf("  ");
         }
+
+        // Note: that is incorrect if we are compiling several methods at the same time.
+        unsigned methodNumber = Compiler::jitTotalMethodCompiled - 1;
```

Comment:
We should create a Compiler member that captures this once and then use that for all output, and use InterlockedIncrement to update this static.